lib/spark.js

```diff
@@ -253,17 +254,23 @@ export default class Spark {
         this.handleRunError(err)
       }
       const duration = Date.now() - started
-      const delay = calculateDelayBeforeNextTask({
-        roundLengthInMs: APPROX_ROUND_LENGTH_IN_MS,
-        maxJitterInMs: MAX_JITTER_BETWEEN_TASKS_IN_MS,
-        maxTasksPerRound: this.#tasker.maxTasksPerRound,
-        lastTaskDurationInMs: duration,
-      })
+      const isHealthy = this.#activity.isHealthy()
+
+      let delay
+      if (isHealthy) {
+        delay = calculateDelayBeforeNextTask({
+          roundLengthInMs: APPROX_ROUND_LENGTH_IN_MS,
+          maxJitterInMs: MAX_JITTER_BETWEEN_TASKS_IN_MS,
+          maxTasksPerRound: this.#tasker.maxTasksPerRound,
+          lastTaskDurationInMs: duration,
+        })
+      } else {
+        delay = OFFLINE_RETRY_DELAY_MS
+      }
 
       if (delay > 0) {
         console.log(
-          'Sleeping for %s seconds before starting the next task...',
-          Math.round(delay / 1000),
+          `[SPARK] ${isHealthy ? 'Online' : 'Offline'} – sleeping for ${Math.round(delay / 1000)}s before next task.`,
```

Comment:
I believe the checker node is already prefixing messages with the module name. Please remove the `[SPARK] ` prefix.

```suggestion
          `${isHealthy ? 'Online' : 'Offline'} – sleeping for ${Math.round(delay / 1000)}s before next task.`,
```